tests/opentracer/test_tracer_asyncio.py

```diff
@@ -4,187 +4,159 @@
-        context switches.
-        """
-        # if multiple coroutines have nested tracing, they must belong
-        # to the same trace
-        @asyncio.coroutine
-        def coro():
-            # another traced coroutine
-            with self.ot_tracer.start_span("coroutine_2"):
-                return 42
-
-        with self.dd_tracer.trace("coroutine_1"):
-            value = yield from coro()
-
-        # the coroutine has been called correctly
-        assert value == 42
-        # a single trace has been properly reported
-        traces = self.ot_tracer._dd_tracer.pop_traces()
-        assert len(traces) == 1
-        assert len(traces[0]) == 2
-        assert traces[0][0].name == "coroutine_1"
-        assert traces[0][1].name == "coroutine_2"
-        # the parenting is correct
-        assert traces[0][0] == traces[0][1]._parent
-        assert traces[0][0].trace_id == traces[0][1].trace_id
+    traces = test_spans.pop_traces()
 
+    assert len(traces) == 1
+    assert len(traces[0]) == 1
+    assert traces[0][0].name == "coroutine"
 
-@pytest.mark.skipif(CONTEXTVARS_IS_AVAILABLE, reason="only applicable to legacy asyncio provider")
-class TestUtilsAsyncio(object):
-    """Test the util routines of the opentracer with asyncio specific
-    configuration.
+
+@pytest.mark.asyncio
+async def test_trace_multiple_coroutines(ot_tracer, test_spans):
+    # if multiple coroutines have nested tracing, they must belong
+    # to the same trace
+
+    async def coro():
+        # another traced coroutine
+        with ot_tracer.start_active_span("coroutine_2"):
+            return 42
+
+    with ot_tracer.start_active_span("coroutine_1"):
+        value = await coro()
+
+    # the coroutine has been called correctly
+    assert value == 42
+    # a single trace has been properly reported
+    traces = test_spans.pop_traces()
+    assert len(traces) == 1
+    assert len(traces[0]) == 2
+    assert traces[0][0].name == "coroutine_1"
+    assert traces[0][1].name == "coroutine_2"
+    # the parenting is correct
+    assert traces[0][0] == traces[0][1]._parent
+    assert traces[0][0].trace_id == traces[0][1].trace_id
+
+
+@pytest.mark.asyncio
+async def test_exception(ot_tracer, test_spans):
+    async def f1():
+        with ot_tracer.start_span("f1"):
+            raise Exception("f1 error")
+
+    with pytest.raises(Exception):
+        await f1()
+
+    traces = test_spans.pop_traces()
+    assert len(traces) == 1
+    spans = traces[0]
+    assert len(spans) == 1
+    span = spans[0]
+    assert span.error == 1
+    assert span.get_tag("error.msg") == "f1 error"
+    assert "Exception: f1 error" in span.get_tag("error.stack")
+
+
+@pytest.mark.asyncio
+async def test_trace_multiple_calls(ot_tracer, test_spans):
+    ot_tracer._dd_tracer.configure(context_provider=context_provider)
+    # create multiple futures so that we expect multiple
+    # traces instead of a single one (helper not used)
+    async def coro():
```

Comment:
missing newline between nested definition (flake8)

Reply:
I'll blame black not doing its job 😛